pyccel/ast/datatypes.py

```diff
@@ -81,7 +81,37 @@
     "logical" : {
         4  : "C_BOOL"}
 }
-
+iso_c_binding_shortcut = {
+    "integer" : {
+        1  : 'i8',
+        2  : 'i16',
+        4  : 'i32',
+        8  : 'i64',
+        16 : 'i128'}, #no supported yet
+    "real"    : {
+        4  : 'f32',
+        8  : 'f64',
+        16 : 'f128'},
+    "logical" : {
+        4  : "b4"}
+}
+iso_c_bindings = ['C_INT8_T', 'C_INT16_T', 'C_INT32_T', 'C_INT64_T', 'C_INT128',
```

Comment:
I realise this isn't your error, but is just copied from above, but would you mind correcting the literal constant for 128 bit integers. `C_INT128` is wrong. It should be `C_INT128_T` (see [here](https://gcc.gnu.org/onlinedocs/gcc-4.6.3/gfortran/ISO_005fC_005fBINDING.html))

Reply:
Noted! I will fix it. support for C_INT128_T is not  a priority?

Reply:
It is not a priority, but it would be simple to add and is also a relatively simple issue. There isn't an issue open for it, but it could be done at the same time as issue #919 . This could be a good next issue for you, as long as PR #886 is merged before then

Reply:
Great!

Reply:
The short version would therefore be `i128`

Reply:
I think support for a 128-bit integer through __int128 will depend on the target machine.
Should a check about it be performed?
Also, C does not allow the initialization of 128-bit integers, this should be done manually.
- Shall the lhs of the assignment be parsed as  a string?
- How shall this reflect on the generated code? can an assignment be represented as a function call?
- The same questions apply to __float128.